File: src/main/java/com/mini/credit/dto/credit/EcheanceCreditResponse.java
package com.mini.credit.dto.credit;

import com.mini.credit.enums.StatutEcheance;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Builder
public class EcheanceCreditResponse {

    private Long id;
    private Integer numeroEcheance;
    private LocalDate dateEcheance;

    private BigDecimal principalPrevu;
    private BigDecimal interetPrevu;
    private BigDecimal penaliteCumulee;
    private BigDecimal totalPrevu;

    private BigDecimal principalPaye;
    private BigDecimal interetPaye;
    private BigDecimal penalitePayee;
    private BigDecimal totalPaye;

    private BigDecimal resteAPayer;

    private BigDecimal principalRestant;
    private BigDecimal interetRestant;
    private BigDecimal penaliteRestante;

    private LocalDate dateDernierPaiement;
    private StatutEcheance statut;
}

File: frontend/src/app/features/credit/models/echeance-credit-response.ts
import { StatutEcheance } from './enums/statut-echeance.enum';

export interface EcheanceCreditResponse {
  id: number;
  numeroEcheance: number;
  dateEcheance: string;

  principalPrevu: number;
  interetPrevu: number;
  penaliteCumulee: number;
  totalPrevu: number;

  principalPaye: number;
  interetPaye: number;
  penalitePayee: number;
  totalPaye: number;

  resteAPayer: number;

  principalRestant: number;
  interetRestant: number;
  penaliteRestante: number;

  dateDernierPaiement?: string | null;
  statut: StatutEcheance;
}